### PR TITLE
added question and collaborations integration

### PR DESCRIPTION
**Pull request by Manan-Sharma-5**
## Summary by CodeRabbit

- **New Features**
	- Enhanced the survey experience by allowing users to accumulate and submit their answers as they navigate through questions.
	- Introduced functions for better navigation between survey questions, including options to skip and go back.
	- Updated salary filter logic to enforce more realistic salary ranges and provide clearer default values in the UI.
	- Implemented dynamic data fetching for collaborations, improving filtering options based on user input.
	- Added a new screen to display detailed information about specific collaborations.

- **Bug Fixes**
	- Improved logic flow for handling user responses, ensuring answers are tracked and submitted correctly before proceeding.

- **Refactor**
	- Streamlined the `Contracts` component by removing unused imports, enhancing performance and maintainability.
	- Enhanced data structure in the `CollaborationCard` component for better clarity and organization.
	- Refactored the `CollaborationPage` component to improve data handling and clarity.

### CODE REVIEW

**Comment by Manan-Sharma-5** [comment]
[Create script to populate Collaborations and brands and manager collections](https://trello.com/c/B5XHBFah/18-create-script-to-populate-collaborations-and-brands-and-manager-collections)

**Comment by Manan-Sharma-5** [comment]
[Onboard questions -> inputs to be saved on users collections](https://trello.com/c/00fpDl4I/17-onboard-questions-inputs-to-be-saved-on-users-collections)

**Comment by Manan-Sharma-5** [comment]
[Collaborations listing and filtering](https://trello.com/c/LGIZmHxQ/19-collaborations-listing-and-filtering)

File: components/collaboration/index.tsx [hunk 1]
```diff
@@ -1,56 +1,123 @@
...
+
+    setCollabs(collabsWithBrandNames);
+  };
 
   const toggleFilterModal = () => {
     setFilterVisible(!filterVisible);
   };
 
-  const filteredList = Collabs.filter((job) => {
+  console.log(
```
Comment (dev-rathore) [inline components/collaboration/index.tsx:94]: is this console required? We can remove this console if not required to not leak data. Make sure console logs are removed everywhere except backend or firebase.

File: app/(main)/(screens)/collaboration-details/[pageID].tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import CollaborationPage from "@/components/collaboration/CollaborationDetails";
+import { useEffect, useState } from "react";
+import { doc, getDoc } from "firebase/firestore";
+import { useLocalSearchParams } from "expo-router";
+import { FirestoreDB } from "@/shared-libs/utilities/firestore";
+import AppLayout from "@/layouts/app-layout";
+import { ActivityIndicator } from "react-native-paper";
+import { ICollaboration } from "@/shared-libs/firestore/trendly-pro/models/collaborations";
+
+interface Collaboration extends ICollaboration {
```
Comment (dev-rathore) [inline app/(main)/(screens)/collaboration-details/[pageID].tsx:10]: Types can be shifted to types folder - file-name.ts, if these types are related to data not to component.

**Review by dev-rathore — APPROVED** [review]
Left two comments, rest looks good.

**Comment by Manan-Sharma-5** [comment]
[Collaboration Details view](https://trello.com/c/yrt5GLMG/20-collaboration-details-view)

**Pull request merged**